### ISSUE TITLE
jdbi.org doc suggestion: demonstrate @CreateSqlObject

### ISSUE THREAD

**Issue by TheRealMarnes** [jdbi/jdbi#979]
In a personal project, I use the following setup that mimicks DAOs + Services:

```java
@RegisterMapper(Foo.Mapper.class)
public interface FooQueries {
	@SqlUpdate("update foo...")
	void create(@Bind("foo") UUID foo);

	@SqlQuery("...")
	Foo select();

	// etc
}

public interface JdbiFooService extends FooService {
	@CreateSqlObject
	FooQueries fooQueries();
	@CreateSqlObject
	ThingQueries thingQueries();

	@Override
	@Transaction
	default Foo linkAllThingsTo(String name) {
		Foo foo = fooQueries().get(name);
		thingQueries().all().forEach(foo::link);
	}
}

Jdbi.create(ds).useExtension(FooService.class, fooService -> linkAllThingsTo("me"));
```

This way, you get responsibility-separated components that can be composed to make higher level services: usage-agnostic query interfaces, and transactional services that implement usecases built on 1 or more such query interfaces. Jdbi's `sql objects` + `@CreateSqlObject` basically feel like `@Component` + `@Inject/@Autowired` from spring.

The current doc doesn't even mention `@CreateSqlObject` anymore, some code like this could make a good example? Or did I create a monster here? :P

**Comment by stevenschlansker** [comment]
Maybe we could start assembling a samples section, or even a separate page.  This is good information but I think having the main document should stay focused on how to use the library, rather than how to integrate Spring with the library.

I also kinda see `@CreateSqlObject` as a bit of a strange design.  It's definitely useful in this case though.

I wonder if this ties in with #967 - would this be better if we actually were able to do
```java
abstract class FooSuper {
    @Inject
    FooSuper(CollaboratorA a, CollaboratorB b) {
        this.a = a;
    }
}
```

and then have jdbi actually figure out proper injection?  (by delegating to an ioc factory you provide?)

**Comment by TheRealMarnes** [comment]
Integrating with spring wasn"t really my point, I added spring to demonstrate the similarity and ease of composition

**Comment by TheRealMarnes** [comment]
I don't think changing anything or providing more is really necessary. Spring makes enabling ioc on arbitrary classes easy enough, and I'm sure other frameworks do too. I don't see anything wrong with `@CreateSqlObject` either, it's quite a clean solution imo. I just wanted to point out its unfortunate absence from the doc.

**Comment by qualidafial** [comment]
I'm not certain that any of the project maintainer use Spring at their day jobs. So it's unlikely this issue will be resolved unless a PR is contributed by someone outside the project.

**Comment by TheRealMarnes** [comment]
But spring isn't the point... it's to show that Jdbi can instantiate
extensions with some responsibilities delegated to subobjects that are
seamlessly instantiated along, so there's no need to make monolith services.

--
Marnick "with ck, 2 capitals, and an apostrophe" L'Eau

**Comment by TheRealMarnes** [comment]
I've updated the first to be about jdbi only and to have v3 code instead of v2.

**Comment by qualidafial** [comment]
Fixed in #982
